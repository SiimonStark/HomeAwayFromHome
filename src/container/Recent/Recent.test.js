import React from 'react';
import { Recent, mapStateToProps } from './index';
import {shallow} from 'enzyme';

describe('Recent', () => {
  describe('Recent component', () => {
    let wrapper;
    let mockRecent;
    beforeEach(() => {
      mockRecent = { listing: [{ id: 0, title: 'job', post_date: "never", company: { name: 'Place', tagline: 'Work here' } }, { id: 1, title: 'JOB', post_date: "Now", company: { name: 'Other', tagline: 'Dont Work here' } }]}
      wrapper = shallow(<Recent loading={false} recent={mockRecent}/>)
    })
    it('should match snapshot if loading is finished', () => {
      expect(wrapper).toMatchSnapshot()
    })
    it('should match snapshot if loading is NOT finished', () => {
      wrapper = shallow(<Recent loading={true} recent={{}} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('mapStateToProps', () => {
    it('should test state to props', () => {
      // Setup
      const mockState = {
        jobs: {listing: [{ name: 'Company', id: 0 }]},
        loading: false
      }
      const expected = {
        recent: { listing: [{ name: 'Company', id: 0 }] }
      }
      // Execution
      const mappedProps = mapStateToProps(mockState)
      // Expectation
      expect(mappedProps).toEqual(expected)
    })
  })
})
