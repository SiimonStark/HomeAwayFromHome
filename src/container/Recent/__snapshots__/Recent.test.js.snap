// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recent Recent component should match snapshot if loading is NOT finished 1`] = `
<section
  className="Recent"
>
  <Loading />
</section>
`;

exports[`Recent Recent component should match snapshot if loading is finished 1`] = `
<section
  className="Recent"
>
  <div
    className="card-container"
  >
    <article
      className="recent-job"
      key="0"
    >
      <div
        className="recent-header"
      >
        <h3>
          Role: 
          job
        </h3>
        <a>
          Apply Here
        </a>
      </div>
      <div
        className="border"
      />
      <div
        className="recent-content"
      >
        <h5>
          Post Date:(
          never
          )
        </h5>
        <h5>
          Company: 
          Place
        </h5>
        <p>
          Work here
        </p>
      </div>
    </article>
    <article
      className="recent-job"
      key="1"
    >
      <div
        className="recent-header"
      >
        <h3>
          Role: 
          JOB
        </h3>
        <a>
          Apply Here
        </a>
      </div>
      <div
        className="border"
      />
      <div
        className="recent-content"
      >
        <h5>
          Post Date:(
          Now
          )
        </h5>
        <h5>
          Company: 
          Other
        </h5>
        <p>
          Dont Work here
        </p>
      </div>
    </article>
  </div>
</section>
`;
